src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -596,37 +597,32 @@ public void stop(Throwable cause) throws Exception {
      * @return the computer
      * @throws AbortException in case of error.
      */
-    private @CheckForNull Computer getComputer() throws AbortException {
+    private @Nonnull Computer getComputer() throws AbortException {
+        if (computer != null) {
+            return computer;
+        }
+
         String node = null;
-        if (computer == null) {
-            Jenkins j = Jenkins.getInstance();
-            if (j == null) {
-                LOGGER.fine("Jenkins is not running");
-                return null;
-            }
-            for (Computer c : j.getComputers()) {
-                if (c.getChannel() == launcher.getChannel()) {
-                    node = c.getName();
-                    break;
-                }
-            }
+        Jenkins j = Jenkins.getActiveInstance();
 
-            if (node == null) {
-                LOGGER.log(Level.FINE, "could not find comuter for this job");
-                return null;
+        for (Computer c : j.getComputers()) {
+            if (c.getChannel() == launcher.getChannel()) {
+                node = c.getName();
+                break;
             }
+        }
 
-            computer = j.getComputer(node);
-            if (computer == null) {
-                LOGGER.log(Level.FINE, "no such computer {0}", node);
-                return null;
-            }
+        if (node == null) {
+            throw new AbortException("Could not find computer for the job");
+        }
 
-            if (computer.isOffline()) {
-                LOGGER.log(Level.FINE, "{0} is offline", node);
-                return null;
-            }
+        computer = j.getComputer(node);
+        if (computer == null) {
+            throw new AbortException("No such computer " + node);
+        }
 
+        if (computer.isOffline()) {

```

Comment:
I guess that for this case we can not do that check... the only thing I need to have is a reference to the computer and the node... having that if the computer is offline should be handled by the pipeline framework it self. That said I'm removing that check.